modules/openthread/platform/ble.c

```diff
@@ -0,0 +1,459 @@
+#define PLAT_BLE_MSG_DISCONNECT (PLAT_BLE_MSG_CONNECT + 1U)
+
+/* Zephyr Kernel Objects */
+
+static void ot_plat_ble_thread(void *, void *, void *);
+static uint8_t ot_plat_ble_msg_buf[PLAT_BLE_MSG_DATA_MAX];
+
+static K_SEM_DEFINE(ot_plat_ble_init_semaphore, 0, 1);
+static K_SEM_DEFINE(ot_plat_ble_event_semaphore, 0, 10000);
+RING_BUF_DECLARE(ot_plat_ble_ring_buf, PLAT_BLE_RING_BUF_SIZE);
+static K_THREAD_DEFINE(ot_plat_ble_tid, PLAT_BLE_THREAD_STACK_SIZE, ot_plat_ble_thread, NULL, NULL,
+		       NULL, 5, 0, PLAT_BLE_THREAD_DEALY);
+
+/* OpenThread Objects */
+
+static otInstance *ble_openthread_instance;
+
+/* BLE service Objects */
+
+/* forward declaration for callback functions */
+static ssize_t on_receive(struct bt_conn *conn, const struct bt_gatt_attr *attr, const void *buf,
+			  uint16_t len, uint16_t offset, uint8_t flags);
+static void on_cccd_changed(const struct bt_gatt_attr *attr, uint16_t value);
+
+/* Service Declaration and Registration */
+BT_GATT_SERVICE_DEFINE(my_service, BT_GATT_PRIMARY_SERVICE(BT_UUID_TCAT_SERVICE),
+		       BT_GATT_CHARACTERISTIC(BT_UUID_TCAT_SERVICE_RX,
+					      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_WRITE_WITHOUT_RESP,
+					      BT_GATT_PERM_READ | BT_GATT_PERM_WRITE, NULL,
+					      on_receive, NULL),
+		       BT_GATT_CHARACTERISTIC(BT_UUID_TCAT_SERVICE_TX, BT_GATT_CHRC_NOTIFY,
+					      BT_GATT_PERM_READ, NULL, NULL, NULL),
+		       BT_GATT_CCC(on_cccd_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),);
+
+/* Zephyr BLE Objects */
+
+/* forward declaration for callback functions */
+static void connected(struct bt_conn *conn, uint8_t err);
+static void disconnected(struct bt_conn *conn, uint8_t reason);
+static bool le_param_req(struct bt_conn *conn, struct bt_le_conn_param *param);
+static void le_param_updated(struct bt_conn *conn, uint16_t interval, uint16_t latency,
+			     uint16_t timeout);
+
+static struct bt_conn *ot_plat_ble_connection;
+
+static struct bt_conn_cb conn_callbacks = {.connected = connected,
+					   .disconnected = disconnected,
+					   .le_param_req = le_param_req,
+					   .le_param_updated = le_param_updated};
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static const struct bt_data sd[] = {
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL, BT_UUID_16_ENCODE(TOBLE_SERVICE_UUID)),
+};
+
+/* Zephyr BLE Message Queue and Thread */
+
+static bool ot_plat_ble_queue_msg(const uint8_t *aData, uint16_t aLen, int8_t aRssi)
+{
+	otError error = OT_ERROR_NONE;
+	uint16_t len = 0;
+
+	if (aLen <= PLAT_BLE_MSG_DATA_MAX && aData == NULL) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+	k_sched_lock();
+
+	len = sizeof(aLen) + sizeof(aRssi) + ((aLen <= PLAT_BLE_MSG_DATA_MAX) ? aLen : 0);
+
+	if (ring_buf_space_get(&ot_plat_ble_ring_buf) >= len) {
+		ring_buf_put(&ot_plat_ble_ring_buf, (uint8_t*)&aLen, sizeof(aLen));
```

Comment:
```suggestion
		ring_buf_put(&ot_plat_ble_ring_buf, (uint16_t *)&aLen, sizeof(aLen));
```

Reply:
nvm,type cast is ok, my bad

Reply:
This is to silence the compiler warning since `ring_buf_get` takes `const uint8_t*` I have to cast `len`.